kclvm/error/src/diagnostic.rs

```diff
@@ -5,11 +5,23 @@ use std::hash::Hash;
 use crate::{ErrorKind, WarningKind};
 
 /// Diagnostic structure.
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Diagnostic {
     pub level: Level,
     pub messages: Vec<Message>,
     pub code: Option<DiagnosticId>,
+    pub data: Option<serde_json::Value>,
```

Comment:
you don’t need to define the data field in kcl’s diag. It is duplicated with suggested_replacement. As I explained in the step 2, the data field should be in the diag of lsp, not the diag of kcl. You can look up the function, kcl_diag_to_lsp_diags. this function convert KCL Diagnostic to LSP Diagnostics. you can see the data in result is always None. you should completed in here

Reply:
Oh right, storing the data field in KCL diagnostics is redundant. Fixed this in latest commit, and made the relevant changes in the code to utilize ```suggested_replacement``` field for conversion of KCL Diagnostic to LSP Diagnostics.